mlir-tensorrt/executor/lib/Runtime/API/API.cpp

```diff
@@ -463,6 +466,22 @@ void AllocTracker::track(PointerInfo info) {
     map.insert(std::make_pair(info.ptr, std::move(value)));
     return;
   }
+
+  auto existingReferenceCount = map.at(info.ptr)->externalReferenceCount.load();
+
+  // If exisiting value in map is internally maanged and not yet released
+  // internally, we can just increment the reference count for the exisiting
```

Comment:
This comment looks incomplete based on the current implementation. can we instead explain the motivation to increase the reference count rather than explain what code does which seems self-explanatory.